common/src/main/java/org/apache/gravitino/dto/tag/MetadataObjectDTO.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.dto.tag;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import org.apache.gravitino.MetadataObject;
+
+/** Represents a Metadata Object DTO (Data Transfer Object). */
+@EqualsAndHashCode
+public class MetadataObjectDTO implements MetadataObject {
+
+  private String parent;
+
+  private String name;
+
+  @JsonProperty("type")
+  private Type type;
+
+  private MetadataObjectDTO() {}
+
+  @Override
+  public String parent() {
+    return parent;
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  @Override
+  public Type type() {
+    return type;
+  }
+
+  /** @return The full name of the metadata object. */
+  @JsonProperty("fullName")
+  public String getFullName() {
+    return fullName();
+  }
+
+  /**
+   * Sets the full name of the metadata object.
+   *
+   * @param fullName The full name of the metadata object.
+   */
+  @JsonProperty("fullName")
+  public void setFullName(String fullName) {
```

Comment:
Why do we need this method instead of using build? MetadataObjectDTO became a mutable object?

Reply:
It is only for Jackson to use.